procrustes/utils.py

```diff
@@ -145,11 +149,17 @@ def _translate_array(array_a, array_b=None):
     """
     # The mean is strongly affected by outliers and is not a robust estimator for central location
     # see https://docs.python.org/3.6/library/statistics.html?highlight=mean#statistics.mean
-    centroid = np.mean(array_a, axis=0)
+    if weight is not None and check_weight:
+        if weight.ndim != 1:
+            raise ValueError("The weight should be a 1d row vector.")
+        if not (weight >= 0).all():
+            raise ValueError("The elements of the weight should be non-negative.")
+
```

Comment:
I really don't understand what is the added value of `check_weight` argument. I believe this is where `check_weight` is being used, but shouldn't we always check the weights? If not, the weight can be checked and a warning can be raised.

Reply:
Thanks for pointing it out, which makes a lot of sense. The problem has been fixed in cd02b69.